api/repository/add.go

```diff
@@ -0,0 +1,132 @@
+package repository
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/gojekfarm/albatross/pkg/logger"
+
+	"github.com/gorilla/mux"
+)
+
+// AddRequest is the body for PUT request to repository
+// swagger:model addRepoRequestBody
+type AddRequest struct {
+	Name     string `json:"-"`
+	URL      string `json:"url"`
+	Username string `json:"username"`
+	Password string `json:"password"`
+	// example: false
+	ForceUpdate bool `json:"force_update"`
+	// CertFile              string
+	// KeyFile               string
+	// CaFile                string
+	InsecureSkipTLSverify bool `json:"skip_tls_verify"`
+}
+
+type addService interface {
+	Add(context.Context, AddRequest) error
+}
+
+// AddErrorResponse body of non 2xx response
+// swagger:model addRepoErrorResponseBody
+type AddErrorResponse struct {
+	Error string `json:"error"`
+}
+
+// AddOkResponse body of 2xx response
+// swagger:model addRepoOkResponseBody
+type AddOkResponse struct {
+	Message string `json:"message"`
+}
+
+const NAME string = "repository-name"
```

Comment:
I was referring to the variable name, not the value, since it represents the flag name and not the repository name (not explicitly stating the purpose). Also can this be used directly in the func?

Reply:
Okay, from a package perspective, what should repository.NAME signify? Would it signify the name of the repository or the url fragment (here it's URL fragment). The name can be something that signifies that it represents a URL fragment and not that repository name.